### PR TITLE
Backport: Feature/move again (#1070)

### PR DESCRIPTION
**Pull request by xavierh**
* move more utils

* more moved

* fix imports

* do merge

Conflicts:
	src/main/java/com/minecolonies/coremod/colony/Colony.java
	src/main/java/com/minecolonies/coremod/commands/MinecoloniesCommand.java
	src/main/java/com/minecolonies/coremod/commands/generalcommands/RandomTeleportCommand.java
	src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java
	src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIInteract.java
	src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java
	src/main/java/com/minecolonies/coremod/entity/pathfinding/AbstractPathJob.java
	src/main/java/com/minecolonies/coremod/items/ItemSupplyCampDeployer.java
	src/main/java/com/minecolonies/coremod/items/ItemSupplyChestDeployer.java
	src/main/java/com/minecolonies/coremod/network/messages/BuildToolPlaceMessage.java

Closes #
Closes #
Closes #

# Changes proposed in this pull request:
-
-
-

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/items/ItemSupplyChestDeployer.java [hunk 1]
```diff
@@ -1,6 +1,8 @@
 package com.minecolonies.coremod.items;
 
+import com.minecolonies.api.configuration.Configurations;
 import com.minecolonies.api.util.constant.Constants;
+import com.minecolonies.api.configuration.Configurations;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/items/ItemSupplyChestDeployer.java]: MINOR Remove this duplicated import. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 138 issues
* ![MAJOR][MAJOR] 94 major
* ![MINOR][MINOR] 37 minor
* ![INFO][INFO] 7 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [CitizenData.java#L23](https://github.com/Minecolonies/minecolonies/blob/26d221c5112afafcf17d3190ecfca5b0040232c8/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L23): class "CitizenData" has 43 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/26d221c5112afafcf17d3190ecfca5b0040232c8/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.265 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L47](https://github.com/Minecolonies/minecolonies/blob/26d221c5112afafcf17d3190ecfca5b0040232c8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L47): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 44 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L47](https://github.com/Minecolonies/minecolonies/blob/26d221c5112afafcf17d3190ecfca5b0040232c8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L47): class "Colony" has 84 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L663](https://github.com/Minecolonies/minecolonies/blob/26d221c5112afafcf17d3190ecfca5b0040232c8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L663): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L850](https://github.com/Minecolonies/minecolonies/blob/26d221c5112afafcf17d3190ecfca5b0040232c8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L850): Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [Colony.java#L1199](https://github.com/Minecolonies/minecolonies/blob/26d221c5112afafcf17d3190ecfca5b0040232c8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1199): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1200](https://github.com/Minecolonies/minecolonies/blob/26d221c5112afafcf17d3190ecfca5b0040232c8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1200): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1313](https://github.com/Minecolonies/minecolonies/blob/26d221c5112afafcf17d3190ecfca5b0040232c8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1313): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1315](https://github.com/Minecolonies/minecolonies/blob/26d221c5112afafcf17d3190ecfca5b0040232c8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1315): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 137 issues
* ![MAJOR][MAJOR] 94 major
* ![MINOR][MINOR] 36 minor
* ![INFO][INFO] 7 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [CitizenData.java#L23](https://github.com/Minecolonies/minecolonies/blob/594674097f35a972ea895e6b8acd9f7555cda440/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L23): class "CitizenData" has 43 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/594674097f35a972ea895e6b8acd9f7555cda440/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.265 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L47](https://github.com/Minecolonies/minecolonies/blob/594674097f35a972ea895e6b8acd9f7555cda440/src/main/java/com/minecolonies/coremod/colony/Colony.java#L47): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 44 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L47](https://github.com/Minecolonies/minecolonies/blob/594674097f35a972ea895e6b8acd9f7555cda440/src/main/java/com/minecolonies/coremod/colony/Colony.java#L47): class "Colony" has 84 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L663](https://github.com/Minecolonies/minecolonies/blob/594674097f35a972ea895e6b8acd9f7555cda440/src/main/java/com/minecolonies/coremod/colony/Colony.java#L663): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L850](https://github.com/Minecolonies/minecolonies/blob/594674097f35a972ea895e6b8acd9f7555cda440/src/main/java/com/minecolonies/coremod/colony/Colony.java#L850): Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [Colony.java#L1199](https://github.com/Minecolonies/minecolonies/blob/594674097f35a972ea895e6b8acd9f7555cda440/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1199): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1200](https://github.com/Minecolonies/minecolonies/blob/594674097f35a972ea895e6b8acd9f7555cda440/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1200): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1313](https://github.com/Minecolonies/minecolonies/blob/594674097f35a972ea895e6b8acd9f7555cda440/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1313): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1315](https://github.com/Minecolonies/minecolonies/blob/594674097f35a972ea895e6b8acd9f7555cda440/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1315): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1160/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 137 issues
* ![MAJOR][MAJOR] 94 major
* ![MINOR][MINOR] 36 minor
* ![INFO][INFO] 7 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [CitizenData.java#L23](https://github.com/Minecolonies/minecolonies/blob/4462ea82db8e0a291f54d1ceb765d2fc2a4d9e15/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L23): class "CitizenData" has 43 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/4462ea82db8e0a291f54d1ceb765d2fc2a4d9e15/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.265 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L47](https://github.com/Minecolonies/minecolonies/blob/4462ea82db8e0a291f54d1ceb765d2fc2a4d9e15/src/main/java/com/minecolonies/coremod/colony/Colony.java#L47): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 44 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L47](https://github.com/Minecolonies/minecolonies/blob/4462ea82db8e0a291f54d1ceb765d2fc2a4d9e15/src/main/java/com/minecolonies/coremod/colony/Colony.java#L47): class "Colony" has 84 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L663](https://github.com/Minecolonies/minecolonies/blob/4462ea82db8e0a291f54d1ceb765d2fc2a4d9e15/src/main/java/com/minecolonies/coremod/colony/Colony.java#L663): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L850](https://github.com/Minecolonies/minecolonies/blob/4462ea82db8e0a291f54d1ceb765d2fc2a4d9e15/src/main/java/com/minecolonies/coremod/colony/Colony.java#L850): Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [Colony.java#L1199](https://github.com/Minecolonies/minecolonies/blob/4462ea82db8e0a291f54d1ceb765d2fc2a4d9e15/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1199): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1200](https://github.com/Minecolonies/minecolonies/blob/4462ea82db8e0a291f54d1ceb765d2fc2a4d9e15/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1200): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1313](https://github.com/Minecolonies/minecolonies/blob/4462ea82db8e0a291f54d1ceb765d2fc2a4d9e15/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1313): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1315](https://github.com/Minecolonies/minecolonies/blob/4462ea82db8e0a291f54d1ceb765d2fc2a4d9e15/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1315): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

**Pull request merged**